plugins/image-prioritizer/class-image-prioritizer-video-tag-visitor.php

```diff
@@ -35,6 +35,25 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 			return false;
 		}
 
+		// TODO: If $context->url_metric_group_collection->get_element_max_intersection_ratio( $xpath ) is 0.0, then the video is not in any initial viewport and the VIDEO tag could get the preload=none attribute added.
+
+		$reduced_poster_size = $this->reduce_poster_image_size( $context );
+		$preload_poster      = $this->preload_poster_image( $context );
+
+		return $reduced_poster_size || $preload_poster;
+	}
+
```

Comment:
We could reduce some duplication by adding this helper method:

```suggestion
	/**
	 * Gets the poster from the current VIDEO element.
	 *
	 * @since n.e.x.t
	 *
	 * @param OD_Tag_Visitor_Context $context Tag visitor context.
	 * @return non-empty-string|null Poster or null if not defined or is a data: URL.
	 */
	private function get_poster( OD_Tag_Visitor_Context $context ): ?string {
		$poster = trim( (string) $context->processor->get_attribute( 'poster' ) );
		if ( '' === $poster || $this->is_data_url( $poster ) ) {
			return null;
		}
		return $poster;
	}

```

This would simplify the code a bit as follows:

```diff
diff --git a/plugins/image-prioritizer/class-image-prioritizer-video-tag-visitor.php b/plugins/image-prioritizer/class-image-prioritizer-video-tag-visitor.php
index 772c290a..9bfe8a89 100644
--- a/plugins/image-prioritizer/class-image-prioritizer-video-tag-visitor.php
+++ b/plugins/image-prioritizer/class-image-prioritizer-video-tag-visitor.php
@@ -37,10 +37,14 @@ final class Image_Prioritizer_Video_Tag_Visitor extends Image_Prioritizer_Tag_Vi
 
 		// TODO: If $context->url_metric_group_collection->get_element_max_intersection_ratio( $xpath ) is 0.0, then the video is not in any initial viewport and the VIDEO tag could get the preload=none attribute added.
 
-		$reduced_poster_size = $this->reduce_poster_image_size( $context );
-		$preload_poster      = $this->preload_poster_image( $context );
+		$poster = $this->get_poster( $context );
+		if ( null !== $poster ) {
+			$this->reduce_poster_image_size( $poster, $context );
+			$this->preload_poster_image( $poster, $context );
+			return true;
+		}
 
-		return $reduced_poster_size || $preload_poster;
+		return false;
 	}
 
 	/**
@@ -48,19 +52,12 @@ final class Image_Prioritizer_Video_Tag_Visitor extends Image_Prioritizer_Tag_Vi
 	 *
 	 * @since n.e.x.t
 	 *
+	 * @param non-empty-string       $poster  Poster image URL.
 	 * @param OD_Tag_Visitor_Context $context Tag visitor context, with the cursor currently at an embed block.
-	 * @return bool Whether the tag should be tracked in URL metrics.
 	 */
-	private function reduce_poster_image_size( OD_Tag_Visitor_Context $context ): bool {
+	private function reduce_poster_image_size( string $poster, OD_Tag_Visitor_Context $context ): void {
 		$processor = $context->processor;
-
-		// Skip empty poster attributes and data: URLs.
-		$poster = trim( (string) $processor->get_attribute( 'poster' ) );
-		if ( '' === $poster || $this->is_data_url( $poster ) ) {
-			return false;
-		}
-
-		$xpath = $processor->get_xpath();
+		$xpath     = $processor->get_xpath();
 
 		$max_element_width = 0;
 
@@ -76,8 +73,6 @@ final class Image_Prioritizer_Video_Tag_Visitor extends Image_Prioritizer_Tag_Vi
 			$smaller_image_url = wp_get_attachment_image_url( $poster_id, array( (int) $max_element_width, 0 ) );
 			$processor->set_attribute( 'poster', $smaller_image_url );
 		}
-
-		return true;
 	}
 
 	/**
@@ -85,19 +80,12 @@ final class Image_Prioritizer_Video_Tag_Visitor extends Image_Prioritizer_Tag_Vi
 	 *
 	 * @since n.e.x.t
 	 *
+	 * @param non-empty-string       $poster  Poster image URL.
 	 * @param OD_Tag_Visitor_Context $context Tag visitor context, with the cursor currently at an embed block.
-	 * @return bool Whether the tag should be tracked in URL metrics.
 	 */
-	private function preload_poster_image( OD_Tag_Visitor_Context $context ): bool {
+	private function preload_poster_image( string $poster, OD_Tag_Visitor_Context $context ): void {
 		$processor = $context->processor;
-
-		// Skip empty poster attributes and data: URLs.
-		$poster = trim( (string) $processor->get_attribute( 'poster' ) );
-		if ( '' === $poster || $this->is_data_url( $poster ) ) {
-			return false;
-		}
-
-		$xpath = $processor->get_xpath();
+		$xpath     = $processor->get_xpath();
 
 		// If this element is the LCP (for a breakpoint group), add a preload link for it.
 		foreach ( $context->url_metric_group_collection->get_groups_by_lcp_element( $xpath ) as $group ) {
@@ -120,7 +108,5 @@ final class Image_Prioritizer_Video_Tag_Visitor extends Image_Prioritizer_Tag_Vi
 				$group->get_maximum_viewport_width()
 			);
 		}
-
-		return true;
 	}
 }
```